### ISSUE TITLE
New field in order for IOSS reference

### ISSUE THREAD

**Issue by mdelgrange** [PrestaShop/PrestaShop#24979]
#### Is your feature request related to a problem?
Following new EU VAT rules and your blog article, it could be interesting to add a new IOSS field in Back Office order page, mainly in relation to marketplace sales.

Context
- New VAT rules impose marketplaces to collect VAT on specific orders.
- New VAT rules recommand merchant to add IOSS reference on parcel (stamp & electronic) to avoid double VAT by local customs.
- In order to help its merchants, eBay marketplace will add an "IOSS" field in each order.
I guess this will be the same for all marketplace.
> - Empty if TVA has not been collected by marketplace
> - IOSS ref if TVA has been collected by marketplace
- IOSS number sould be used by carriers modules when creating transport label.

#### Describe the solution you'd like
Add a new field in Back Office order page dedicated to VAT collection.
- This field would only be displayed in Back Office, no need for front. Should be think twice for invoice PDF file.
- This field would be freely completed, in the same way as the parcel tracking code.

#### Alternatives you've considered
In its next version, eBay module will add (happen) IOSS number to transaction reference.
This solution is a workaround which will not facilitate modules interoperability ; each marketplace module may choose different implementation.

#### Additional context
If accepted : 
- 202 ecommerce may be ready to do the PR at PS core level.
- eBay module would behave differently depending on PrestaShop version, transaction reference for previous versions, dedicated field for new versions.

**Comment by hibatallahAouadni** [comment]
Hello @mdelgrange 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

_PS: if you create a PR, this feature will be treated as a **priority**._

Thank you

**Comment by mdelgrange** [comment]
@hibatallahAouadni yes, we can do a PR, but i wait Product Team validation before coding.

**Comment by hibatallahAouadni** [comment]
Great :tada: 

Ping @marionf wdyt?

**Comment by marionf** [comment]
Hello @mdelgrange 

Thank you for your motivation on the subject. 
Before developing such a feature on this subject, we should dive into the legal requirements in order to be sufficiently informed to make the right functional decisions.
Unfortunately, we don't currently have the necessary legal resources to allow us to dive into this subject in order to make the right decision and to be able to write and send you the specifications necessary for development.

I will be sure to get back to you when we will have the resources :)

**Comment by delete13** [comment]
Salut @mdelgrange ;)

Hello @marionf , 

You are saying; 

> Unfortunately, we don't currently have the necessary legal resources to allow us to dive into this subject in order to make the right decision and to be able to write and send you the specifications necessary for development.

> I will be sure to get back to you when we will have the resources :)

Seriously, you are not aware of the European e-commerce regulations ?

Consequently, you systematically shift the responsibility to the agencies.

Here is a good example when we want to post a module on PrestaShop Addons; 

![image](https://user-images.githubusercontent.com/16521037/125889227-1de96cda-4680-41d1-b2b3-6114bf331c60.png)

To be constructive, in this topic, as long as you won't follow the regulations, it will create a bad customer experience. 

End customers will have to suffer because they will have to manage custom's duties, VAT, etc, and those constraints are making marketplaces such as Amazon successful, as they are managing all for their customers; duties, VAT, etc. 

Indeed, the IOSS is not a big deal, this is a simple additional parameter in PrestaShop and a field on the invoice. 

I do agree 100% with Matthieu Delgrange, and, you, PrestaShop, must work on the legal requirement for merchants in Europe because you are claiming that you are the most important e-commerce platform in Europe.

In addition, your agencies are losing merchants because the platform doesn't comply with the regulations.

Can we build together an action plan? 

Best regards,
Olivier Baquet, 
Common-Services

**Comment by ldeprestashop** [comment]
Hello @delete13 

Prestashop is an open-source project.

We (Prestashop company) participate strongly to this project but we don't have ressources to work on vat now. 

Prestashop dev team is 4,5 equivalent full time dev. 
it's a very small team who work hard to improve the core (plus 60 native modules).

We (product management) want work on compliance : it will be a future project. 
But we have not time now and can't work seriously without legal skill : that's why we are looking for partners for work on this topic.

**Comment by matks** [comment]
Also

> you, PrestaShop, must work on the legal requirement for merchants in Europe

There are people, merchants and agencies, that use PrestaShop in a lot of parts outside of Europe. South America. North America. China. India. So when you say "you, PrestaShop, must work on the legal requirement for merchants in Europe" you are asking us to ignore the needs of these people, and prefer your needs instead. Because the PrestaShop community from India and South America does not care about Europe regulations being implemented in PrestaShop.

This is why it's better to implement country or zone legal restrictions inside a module. This allows people who need the module to install it, and the other people don't install it and dont have unnecessary code inside the software, just like the GDPR module. I dont think any country or zone regulation should go in the Core (so ecotax is a mistake I believe).

**Comment by ldeprestashop** [comment]
Hello @mdelgrange.

We can't validate this type of feature without legal legal specifications. 

Does your company have legal partner ? 
If yes you could work with it to provide functional specification (first step before lauching developement). 

Once the specifications are validated, no problem to launch the PR.

**Comment by Hlavtox** [comment]
This can be EASILY implemented via a module. :-)
